test/lib/jdk/test/lib/security/NSSArtifactFetcher.java

```diff
@@ -0,0 +1,141 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.test.lib.security;
+
+import jdk.test.lib.Platform;
+import jdk.test.lib.security.artifacts.ThirdPartyArtifacts;
+import jtreg.SkippedException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Properties;
+import java.util.stream.Stream;
+
+public class NSSArtifactFetcher {
+
+    public static final String DEFAULT_NSS_LIBRARY =  "softokn3";
+
+    public static Path fetchNssLib(String osId, Path libraryName) {
+
+        final Class<?> nssLibClass = getNssLibClass(osId);
+        if(nssLibClass == null){
+            throw new RuntimeException("Platform not recognised"); // should never happen
+        }
+
+        return fetchNssLib(nssLibClass, libraryName);
+    }
+
+    public static Class<?> getNssLibClass(String osId){
+        switch (osId) {
+            case "Windows-amd64-64":
+                return ThirdPartyArtifacts.WINDOWS_X64.class;
+
+            case "MacOSX-x86_64-64":
+                return ThirdPartyArtifacts.MACOSX_X64.class;
+
+            case "MacOSX-aarch64-64":
+                return ThirdPartyArtifacts.MACOSX_AARCH64.class;
+
+            case "Linux-amd64-64":
+                if (Platform.isOracleLinux7()) {
+                    throw new SkippedException("Skipping Oracle Linux prior to v8");
+                } else {
+                    return ThirdPartyArtifacts.LINUX_X64.class;
+                }
+
+            case "Linux-aarch64-64":
+                if (Platform.isOracleLinux7()) {
+                    throw new SkippedException("Skipping Oracle Linux prior to v8");
+                } else {
+                    return ThirdPartyArtifacts.LINUX_AARCH64.class;
+                }
+            default:
+                return null;
+        }
+    }
+
+    public static String getNSSLibDir(String library) throws Exception {
+        Path libPath = getNSSLibPath(library);
+
+        String libDir = String.valueOf(libPath.getParent()) + File.separatorChar;
+        System.out.println("nssLibDir: " + libDir);
+        System.setProperty("pkcs11test.nss.libdir", libDir);
+        return libDir;
+    }
+
+    public static Path getNSSLibPath(String library) throws Exception {
+        String osid = getOsId();
+        Path libraryName = Path.of(System.mapLibraryName(library));
+        Path nssLibPath = fetchNssLib(osid, libraryName);
+        if (nssLibPath == null) {
+            throw new SkippedException("Warning: unsupported OS: " + osid
+                                       + ", please initialize NSS library location, skipping test");
+        }
+        return nssLibPath;
+    }
+
+    public static String getOsId() {
```

Comment:
does this method and the one that returns the @Artifact class need to be public?